src/github/middleware.ts

```diff
@@ -103,12 +104,23 @@ export default (
 			webhookReceived
 		};
 
-		// For all micros envs log the paylaod. Omit from local to reduce noise
-		const loggerWithWebhookParams = process.env.MICROS_ENV
-			? logger.child(webhookParams)
-			: logger.child(omit(webhookParams, "payload"));
+		const propageRequestId = await booleanFlag(BooleanFlags.PROPAGATE_REQUEST_ID, true);
+
+		if (propageRequestId) {
+			// For all micros envs log the paylaod. Omit from local to reduce noise
+			const loggerWithWebhookParams = process.env.MICROS_ENV
+				? context.log.child({name: LOGGER_NAME, ...webhookParams})
```

Comment:
The root of the PR: use Probot's logger instead of creating a new one, thus propagating request id